src/mon/MonmapMonitor.cc

```diff
@@ -676,7 +676,25 @@ bool MonmapMonitor::prepare_command(MonOpRequestRef op)
             << "persistent = " << pending_map.persistent_features
             // output optional nevertheless, for auditing purposes.
             << ", optional = " << pending_map.optional_features << dendl;
-    
+
+  } else if (prefix == "mon set-rank") {
+    string name;
+    int64_t rank;
+    if (!cmd_getval(g_ceph_context, cmdmap, "name", name) ||
+	!cmd_getval(g_ceph_context, cmdmap, "rank", rank)) {
+      err = -EINVAL;
+      goto reply;
+    }
+    int oldrank = pending_map.get_rank(name);
+    if (oldrank < 0) {
+      ss << "mon." << name << " does not exist in monmap";
+      err = -ENOENT;
+      goto reply;
+    }
+    err = 0;
+    pending_map.set_rank(name, rank);
```

Comment:
nit, might want to check the return val of `pending_map.set_rank(name, rank)`, like
```c++
switch (int ret = pending_map.set_rank(name, rank); ret) {
case -ENOENT:
  ss << "mon." << name << " does not exist in monmap";
  err = -ENOENT;
  break;
case -EINVAL;
  err = -EINVAL;
  break;
case 0:
  pending_map.last_changed = ceph_clock_now();
  propose = true;
  err = 0;
  break;
default:
  // already set
  err = 0;
  break;
}

Reply:
I didn't break it out this way because it's operating on pending_map, and we might have a no-op on pending_map that is still waiting for the proposal to go through.  It's simpler to do a proposal even for the no-op case than repeat all of the checks for both committed and pending states.